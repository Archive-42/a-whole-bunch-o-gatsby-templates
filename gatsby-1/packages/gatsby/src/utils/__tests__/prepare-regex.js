const { prepareRegex } = require(`../prepare-regex`)

describe(`Prepare regex for filtering`, () => {
  it(`handles simple regex`, () => {
    expect(prepareRegex(`/blue/`)).toMatchSnapshot()
  })

  it(`handles flags regex`, () => {
    expect(prepareRegex(`/blue/i`)).toMatchSnapshot()
  })

  it(`handles slashes`, () => {
    expect(prepareRegex(`/bl/ue/i`)).toMatchSnapshot()
  })

  it(`handles escape sequences`, () => {
    const expected = /^\w+\d{2}\.$/
    expect(prepareRegex(`/^\\w+\\d{2}\\.$/`)).toEqual(expected)
  })

  it(`handles regex string passed as graphql arg`, async () => {
    const {
      GraphQLSchema,
      GraphQLObjectType,
      GraphQLString,
      graphql,
    } = require(`graphql`)
    const QueryType = new GraphQLObjectType({
      name: `Query`,
      fields: {
        regex: {
          type: GraphQLString,
          args: {
            regex: { type: GraphQLString },
          },
          resolve: (source, args) => args.regex,
        },
      },
    })
    const schema = new GraphQLSchema({ query: QueryType })
    /* prettier-ignore */
    const results = await graphql(
      schema,
      `
        {
          regex(regex: "/\\\\w+/")
        }
      `
    )
    expect(results.errors).toBeUndefined()
    expect(results.data).toEqual({ regex: `/\\w+/` })

    const expected = /\w+/
    expect(prepareRegex(results.data.regex)).toEqual(expected)
  })
})
