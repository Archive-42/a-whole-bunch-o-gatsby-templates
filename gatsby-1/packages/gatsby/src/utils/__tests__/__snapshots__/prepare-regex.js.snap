// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prepare regex for filtering handles flags regex 1`] = `/blue/i`;

exports[`Prepare regex for filtering handles simple regex 1`] = `/blue/`;

exports[`Prepare regex for filtering handles slashes 1`] = `/bl\\\\/ue/i`;
